### ISSUE TITLE
Missing DLLs in the build folder

### ISSUE THREAD

**Issue by jlaumon** [arc80/FlapHero#4]
Hey again :) 

When building on Windows, several required DLLs are not copied to the exe folder:
- bz2.dll
- Irrlicht.dll
- jpeg62.dll
- libpng16.dll

Not completely sure why they are needed, I think I saw that at least Irrlicht was installed at the same time as assimp by vcpkg.

**Comment by preshing** [comment]
Hello again to you too!

You're using VS2019 and building for x64, right?

What are the contents of your `<workspace>\data\build\glfwFlap\build\Debug` folder? (Assuming you build the Debug configuration.)

Do you have your own installation of vcpkg.exe somewhere on your system? What happens when you type `where vcpkg` at the command prompt?

**Comment by jlaumon** [comment]
Indeed VS2019 x64. Here's the content of Debug (but the issue is the same for RelWithAsserts):

```
assimp-vc142-mt.dll
flapGame.lib
flapGame.pdb
glad.lib
glad.pdb
glfw3.dll
glfwFlap.exe
glfwFlap.ilk
glfwFlap.pdb
image.lib
image.pdb
math.lib
math.pdb
runtime.lib
runtime.pdb
soloud.lib
soloud.pdb
zlib1.dll
```

I followed the build instructions for FlapHero that suggested to install vcpkg in the data folder, and `where vcpkg` finds it there (it's also the only one in my PATH as far as I can tell).

**Comment by jlaumon** [comment]
Looks like all the other missing DLLs are dependencies of irrlicht, and I've just seen this https://github.com/microsoft/vcpkg/issues/14015 which suggest it's possible to remove irrlicht (but I'm clueless about vcpkg/CMake so not sure if it's easy to do 😄 ).

**Comment by preshing** [comment]
When I developed Flap Hero, running `vcpkg install assimp:x64-windows` gave you version `5.0.1`. Now it gives you version `5.0.1#4` which seems to bring in a bunch of additional dependencies (as reported by `vcpkg depend-into assimp`) that it didn't have before.

Old assimp dependencies: `minizip, rapidjson, zlib`
New assimp dependencies: `irrlicht, kubazip, minizip, poly2tri, polyclipping, rapidjson, stb, utfcpp, zlib`

Possible fixes:

1. See if the "assimp portfile" (mentioned in the link you shared) can be locally modified somehow before running  `vcpkg install assimp:x64-windows`. I'm not sure if it's possible to do something like that locally or if the portfile is centralized.
2. Look for ways to install older versions of vcpkg packages (to "lock in" to a specific version of assimp that is known to work). [This post](https://medium.com/@allenkuo/using-specified-lib-version-in-vcpkg-ae9ecba3d768) suggests that it can be done by checking out an older commit of vcpkg itself. Not sure if there are other ways.
3. Look for a way to teach PlyTool about all the .libs and .dlls used by a vcpkg package.
4. Just add the new required DLLs to the assimp.vcpkg extern provider in FlapHero.

I've implemented option 4 locally and it seems to work. Guess I'll submit it. This is likely to break again if the vcpkg maintainers decide to change the assimp package again. I'll explore options 1 - 3 and see if there are any viable long-term strategies to avoid issues like this in the future.
